apps/src/javalab/JavalabConsole.jsx

```diff
@@ -61,11 +63,44 @@ class JavalabConsole extends React.Component {
     commandHistory: new CommandHistory(),
   };
 
+  hasCompilationError = log => {
+    return log.text && log.text.includes(javalabMsg.compilerError());
+  };
+
+  hasCompilationSucces = log => {
+    return log.text && log.text.includes(javalabMsg.compilationSuccess());
+  };
+
+  checkForCompilationError = consoleLogs => {
```

Comment:
(Feel free to ignore this suggestion since this is prototype code)
Instead of checking all the logs, could we instead do work in [onOutputMessage](https://github.com/code-dot-org/code-dot-org/blob/f2142338b9edc6558013916e55faca9502c80c36/apps/src/javalab/Javalab.js#L501), and set `hasCompilationError` to `true` if we just got a compilation error message, and to `false` if we just got a success message? We could also do this as a [thunk](https://redux.js.org/usage/writing-logic-thunks#writing-thunks) that we call instead of `appendOutputLog` that will dispatch updates for both the console log object and do the check for a compilation error.

Reply:
Great idea. I moved the compilation error check to onOutputMessage which is much simpler! Thanks!